ipaserver/install/bindinstance.py

```diff
@@ -423,17 +423,15 @@ def zonemgr_callback(option, opt_str, value, parser):
             encoding = getattr(sys.stdin, 'encoding', None)
             if encoding is None:
                 encoding = 'utf-8'
-            value = value.decode(encoding)
+            if not isinstance(value, unicode):
+                value = value.decode(encoding)
+
             validate_zonemgr_str(value)
         except ValueError as e:
             # FIXME we can do this in better way
             # https://fedorahosted.org/freeipa/ticket/4804
             # decode to proper stderr encoding
-            stderr_encoding = getattr(sys.stderr, 'encoding', None)
-            if stderr_encoding is None:
-                stderr_encoding = 'utf-8'
-            error = unicode(e).encode(stderr_encoding)
-            parser.error("invalid zonemgr: " + error)
+            parser.error("invalid zonemgr: " + unicode(e))
```

Comment:
If I understood correctly what you're saying, this would not help if `error = unicode(e)`, because in python3 is not possible to "add" str +  bytes.

*Python 3*:  
```python
[root@server vagrant]# python3.5
Python 3.5.3 (default, May 10 2017, 15:05:55)
>>> b"a" + str("b")
TypeError: can't concat bytes to str
>>> b"a" + "b"
TypeError: can't concat bytes to str
```

*Python 2.7*
```python
[root@server vagrant]# python2.7
Python 2.7.13 (default, May 10 2017, 20:04:28)
>>> b"a" + unicode("b")
u'ab'
>>> b"a" + str("b")
'ab'
>>> b"a" + "b"
'ab'
```

This `parser.error("invalid zonemgr: " + unicode(e))` black magic works (to py3 and py2), because at the begging of file we have (line 62):
```python
if six.PY3:
    unicode = str
```

I've tested with the error described in ticket 4804, and with this new implementation it works:
```bash
[root@server vagrant]# python2  /usr/sbin/ipa-dns-install --zonemgr=TÃªko@example.com
ipa-dns-install: error: invalid zonemgr: domain name 'TÃªko.example.com' should be normalized to: tãako.example.com\

[root@server vagrant]# python3  /usr/sbin/ipa-dns-install --zonemgr=TÃªko@example.com
ipa-dns-install: error: invalid zonemgr: domain name 'TÃªko.example.com' should be normalized to: tãako.example.com
```